include/lsst/afw/detection/Psf.h

```diff
@@ -29,6 +29,7 @@
 #include <memory>
 
 #include "lsst/daf/base.h"
+#include "lsst/utils/Cache.h"
```

Comment:
As discussed on the `utils` ticket, I'd prefer to forward-declare `Cache`, have `Psf` hold it by `unique_ptr`, and move this include to the .cc file.